service/lib/agama/registration.rb

```diff
@@ -333,42 +333,35 @@ def handle_ssl_error(_error, certificate_imported)
       return false if certificate_imported
 
       cert = SSL::Errors.instance.ssl_failed_cert
-      error_code = SSL::Errors.instance.ssl_error_code
-
-      if cert
-        SSL::Storage.instance.fingerprints.each do |fp|
-          # import certificate if it match predefined fingerprint
-          return true if cert.match_fingerprint?(fp)
-        end
-      end
-
-      if cert && SSL::ErrorCodes::IMPORT_ERROR_CODES.include?(error_code)
-        error_msg = format(
-          _("Secure Connection Error for %{url}: %{error}."),
-          url:     registration_url || "https://scc.suse.com",
-          error:   SSL::ErrorCodes::OPENSSL_ERROR_MESSAGES[error_code],
-        )
-        cert_details = format(
-          _("Certificate details %{details}."),
-          details: SSL::CertificateDetails.new(cert).summary
-        )
+      return false unless cert
 
-        message = "<p>#{error_msg}</p><p>#{cert_details}</p>"
+      # Import certificate if it matches predefined fingerprint.
+      return true if SSL::Storage.instance.fingerprints.any? { |f| cert.match_fingerprint?(f) }
 
-        question = Agama::Question.new(
-          qclass:         "registration.certificate",
-          text:           message,
-          options:        [:Import, :Abort],
-          default_option: :Abort
-        )
+      error_code = SSL::Errors.instance.ssl_error_code
+      return false unless SSL::ErrorCodes::IMPORT_ERROR_CODES.include?(error_code)
+
+      details = SSL::CertificateDetails.new(cert)
+
+      question = Agama::Question.new(
+        qclass:         "registration.certificate",
+        text:           _("Secure connection error. Import certificate?"),
+        options:        [:Import, :Abort],
+        default_option: :Abort,
+        data: {
+          url: registration_url || "https://scc.suse.com",
+          error: SSL::ErrorCodes::OPENSSL_ERROR_MESSAGES[error_code],
+          subject: details.subject,
+          issuer: details.issuer,
+          summary: details.summary,
+          fingerprints: SSL::Storage.instance.fingerprints
```

Comment:
this is wrong. It is fingerprints that are predefined in profile. real fingerprints are in `cert.fingerprint` method - https://github.com/agama-project/agama/pull/2270/files#diff-9290e8a5ea2fd76513bbb795cc65cf885f7a6f7bf8802408be0dc41fba6c07f7R167
cert details also using the cert one https://github.com/agama-project/agama/pull/2270/files#diff-d875c71d1027f7a5456ca7e908bfd00a586285e5b9a9283d9657c3e569d8b77dR30 ( both SHA1 and SHA256 )